### PR TITLE
Add CupertinoSwitch

### PR DESCRIPTION
**Pull request by abarth**
This patch contains a first draft of an iOS-style switch. We'll likely need to
tweak the constants upon closer study, but the basics are here.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/painting/box_painter.dart [hunk 1]
```diff
@@ -526,13 +526,18 @@ class BoxShadow {
   /// The amount the box should be inflated prior to applying the blur.
   final double spreadRadius;
 
+  /// Converts a blur radius in pixels to sigmas.
+  ///
+  /// See SkBlurMask::ConvertRadiusToSigma().
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/box_painter.dart]: these should be double slash, not triple slash.

File: packages/flutter/lib/src/painting/box_painter.dart [hunk 2]
```diff
@@ -526,13 +526,18 @@ class BoxShadow {
   /// The amount the box should be inflated prior to applying the blur.
   final double spreadRadius;
 
+  /// Converts a blur radius in pixels to sigmas.
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/box_painter.dart]: You probably still want the `See the sigma argument to [MaskFilter.blur].` part.

File: packages/flutter/test/cupertino/switch_test.dart [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          );
+        },
+      ),
+    );
+
+    expect(value, isFalse);
+    await tester.tap(find.byKey(switchKey));
+    expect(value, isTrue);
+  });
+}
```
Comment (Hixie) [inline packages/flutter/test/cupertino/switch_test.dart]: This might be a good reason to go down the reference test yak shave...

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/test/cupertino/switch_test.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/test/cupertino/switch_test.dart]: I see what you're saying (that this test is really shallow), but I'm not quite ready for that yet.  It's also likely we'll need to tweak the parameters for the switch a bunch before perfect...

Comment (Hixie) [inline packages/flutter/test/cupertino/switch_test.dart]: :-)

**Pull request merged**